src/components/Navbar.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useRef, useEffect } from 'react';
+
+const Navbar = () => {
+  const [dropdown, setDropdown] = useState(false);
+  const ref = useRef();
+  useEffect(() => {
+    const handler = (event) => {
+      if (dropdown && ref.current && !ref.current.contains(event.target)) {
+        setDropdown(false);
+      }
+    };
+    document.addEventListener('mousedown', handler);
+    return () => {
+      document.removeEventListener('mousedown', handler);
+    };
+  }, [dropdown]);
+
+  return (
+    <nav className="navbar">
+      <ul>
+        <li>Home</li>
+        <li>About</li>
+        <li ref={ref}>
+          <button type="button" className="dropdown_button" onClick={() => setDropdown((prev) => !prev)}>
+            Services
```

Comment:
Hi @HFG43 😃, great looking component!

**[OPTIONAL]**

I would recommend creating a function within this component
to be in charge of toggling the `setDropdown` previous value.

e.g.
```js
const toggleDropdown = () {
    setDropdown((prev) => !prev);
}
```
❗ By extracting the click event handler into a separate function 
 (toggleDropdown in this case),
 you achieve better code organization and separation of concerns. It 
 can also make the code more
 readable and easier to understand, especially if the handler logic  
 becomes more complex in the future 🚀.

Ultimately, the decision to use an inline arrow function or a separate handler function depends
 on your personal preference and the specific requirements of your codebase. Both approaches
 are valid and can be used interchangeably 😃.